websites/A/Armorgames/presence.ts

```diff
@@ -0,0 +1,112 @@
+      }
+    } else if (page == "/community") {
+      presenceData.details = `Viewing the forums`;
+    } else if (page.includes("/forum/")) {
+      title = document.querySelector("#content-canvas > main > h1");
+      presenceData.details = `Viewing Forum:`;
+      presenceData.state = title.textContent.replace("Forums → ", "");
+      presenceData.buttons = [{label: "View Forum", url: window.location.href}];
+    } else if (page.includes("thread")) {
+      title = document.querySelector("#thread-title");
+      presenceData.details = `Reading Forum Thread:`;
+      presenceData.state = title.textContent;
+      presenceData.buttons = [{label: "View Thread", url: window.location.href}];
+    } else if (page.includes("/friends/")) {
+      title = document.querySelector("#main > h2");
+      if (title.textContent == "Your Friends") {
+        presenceData.details = `Viewing their friends`;
+        presenceData.buttons = [{label: "View Friends", url: window.location.href}];
+      } else {
+        presenceData.details = `Viewing the friends of:`;
+        presenceData.state = title.textContent;
+        presenceData.buttons = [{label: "View Friends", url: window.location.href}];
+      }
+    } else if (page.includes("/user/")) {
+      title = document.querySelector("#user-data > h1");
+      presenceData.details = `Viewing the profile of:`;
+      presenceData.state = title.textContent;
+      presenceData.buttons = [{label: "View Profile", url: window.location.href}];
+    } else if (page.includes("/author/")) {
+      title = document.querySelector("#categorylisting > h2 > a");
+      presenceData.details = `Viewing Games Made By:`;
+      presenceData.state = title.textContent;
+      presenceData.buttons = [{label: "View Profile", url: window.location.href}];
+    } else if (page == "/register" || page == "/register/") {
+      presenceData.details = "Registering";
+    } else if (page == "/settings/general") {
+      presenceData.details = "Viewing:";
+      presenceData.state = "General Settings";
+    } else if (page == "/settings/friends") {
+      presenceData.details = "Viewing:";
+      presenceData.state = "Their Friends";
+    } else if (page == "/settings/quests") {
+      presenceData.details = "Viewing:";
+      presenceData.state = "Quests";
+    } else if (page == "/settings/favs") {
+      presenceData.details = "Viewing:";
+      presenceData.state = "Favs";
+    } else if (page == "/settings/contact") {
+      presenceData.details = "Viewing:";
+      presenceData.state = "Contact Settings";
+    } else if (page == "/settings/accounts") {
+      presenceData.details = "Viewing:";
+      presenceData.state = "Soicial Settings";
+    } else if (page == "/settings/purchases") {
+      presenceData.details = "Viewing:";
+      presenceData.state = "My Purchases";
+    } else if (page == "/settings/password") {
+      presenceData.details = "Viewing:";
+      presenceData.state = "Password Settings";
```

Comment:
Instead of setting details every time maybe just check on what sites it shouldn't be and then set it? :)